doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,150 @@
+  a single column. `GROUP BY (year, month)` is more awkward.
+
+The calendar table will usually be joined with Simple's application tables to populate the denormalized reporting
+tables. The calendar table will be implemented as a non-materialized database view, since it is fairly small and not
+expensive to compute when needed.
+
+### Reports
+
+Since reporting queries will be simplified, we can start using Metabase for new reports. Metabase is ideal for quick
+prototyping and validation of reports with stakeholders because:
+
+* Easy-to-use interface
+* Easy to grant access to stakeholders, and controlled access to ourselves if necessary
+* Dynamic queries (eg. pick your own date range)
+
+Once reports are validated, we may want to move them in-house into the Simple Server application/dashboard. The benefits
+of doing so would be:
+
+* Better test coverage
+* Less fragmentation
+
+In any case, the long-term location of the reports themselves will not greatly impact the reporting table design
+described here.
+
+## Alternatives and constraints
+
+Apart from the proposed approach of reporting tables as database views, there are some alternatives that were
+considered.
+
+### Reporting tables in a separate database
+
+We could continue to generate the reporting tables by hand, but place them in a separate database. This would greatly
+alleviate performance problems. However, database performance is a secondary concern. Furthermore, the proposed approach
+will still set us up nicely to extract the reporting tables to a separate database in the future if necessary.
+
+### Rollup summary tables
+
+Rollup summary tables were proposed in [an earlier ADR](011-rollup-summary-tables-for-dashboards.md). The crux of the
+approach is to incrementally write data to these tables as application transactions occur. Eg. update a few rows in the
+"latest BPs per month" table when a new BP is recorded. The main drawback of this approach is that it's tough to iterate
+on the schemas, since every change would require a backfill of data from the beginning of time.
+
+### Third-party data warehouse like Looker or Tableau
+
+There's a high up-front cost and complexity to using a third-party tool. Additionally, there may be hidden constraints
+in such tools that we don't want to deal with while we're in an iterative stage of this effort.
+
+## Status
+
+Proposed
+
+## Consequences
+
+* We will have more nightly cron tasks to refresh the reporting views. We need to monitor the performance of these
+  refreshes closely to identify when database views are no longer a viable option for us.
+* We can start thinking of migrating existing dashboard reports to consume data from the reporting tables. Some good
+  candidates to start migrating could possibly be:
+  * Registrations: It is a simple report and easy to verify
+  * Lost-to-follow-up breakdowns: It tests the more complex aspects of the reporting tables, but is fairly isolated.
```

Comment:
I feel (not strongly) like this should be an _intended_ consequence, and should move into the decision section. But it's cool if we want to try it out first, and then commit to it ✌️